Ports/Android/src/com/codename1/impl/android/CodenameOneView.java

```diff
@@ -472,6 +472,32 @@ public boolean onTouchEvent(MotionEvent event) {
         return consumeEvent;
     }
 
+    private Component safetyGetComponentAt(final Container root, final int x, final int y){
+        if (!root.isVisible() || !root.contains(x, y)) {
+            return null;
+        }
+
+        List<Component> children = root.getChildrenAsList(true);

```

Comment:
This is a bit expensive as it allocates an array list for every entry. Since drag events can fire very quickly and impact performance we need to use fast/low level APIs that ideally wouldn't trigger allocations. 

I suggest doing something like:
```java
int componentCount = getComponentCount();
int staringPoint = 0;
int endPoint = componentCount;
int direction = 1;
if(root.getLayout() instanceof LayeredLayout){
    direction = -1;
    endPoint = -1;
    staringPoint = componentCount - 1;
}
for(int iter = startingPoint ; iter != endPoint ; iter += direction) {
     Component currComponent = root.getComponentAt(iter);
    // ...
}
```